### ISSUE TITLE
Support editing already existing annotations

### ISSUE THREAD

**Issue by jointri** [mozilla/pdf.js#15403]
Attach (recommended) or Link to PDF file here:
[dummy.pdf](https://github.com/mozilla/pdf.js/files/9497274/dummy.pdf)

Configuration:
- Tested in firefox beta and nightly.
- Microsoft Windows 10.
- Whatever pdf.js version comes with firefox beta and nightly.
- No, not an extension.

Steps to reproduce the problem:
1. Add a text annotation (Same thing applies if you used the ink annotation, but for this example we shall just use the text annotation) to the attached dummy.pdf (It can be anything, like foo, bar, foobar, and whatever else you want)
2. Save the PDF under a different name (It could be dummytwo.pdf it doesn't matter as long as its a separate file)

What is the expected behavior? (add screenshot)
I should be able to select the saved annotation (this issue applies to any PDF with saved annotations as well, not just made in pdf.js) and delete it or modify it further.
![dummypdf](https://user-images.githubusercontent.com/81046576/188646844-79248563-7b67-4a69-bc98-5380de73a7b8.png)

What went wrong?
I cannot delete it or edit it. As indicated by the screenshot, it acknowledges the existence of the annotation, but I can't delete it or modify it. I should be able to delete annotations or edit annotations made in pdf.js as well as any other PDF reader/writer that can make annotations.

**Comment by jointri** [comment]
I should note that it's not just deleting I cannot do, but I cannot also modify saved annotations.

**Comment by SebCorbin** [comment]
The format from `pdfPage.getAnnotations()` is not the same as in `pdfDocument.annotationStorage`

After going through the source code, the `AnnotationFactory.saveNewAnnotations()` takes care of converting added annotations from `annotationStorage` to PDFAnnotation, but I see no way of doing the inverse through the current API.

**Comment by marco-c** [comment]
Can be closed after #16525 is merged.

**Comment by DDHSchmidt** [comment]
If I read/tested this correctly this was implemented/corrected for text annotations exclusively.
Would it be possible to also enable this for ink annotations since PDF.js is also capable of editing those?

**Comment by marco-c** [comment]
Presumably yes, could you file a new issue?
